reciprocalspaceship/algorithms/intensity.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+import reciprocalspaceship as rs
+
+
+def compute_intensity_from_structurefactor(
+    ds,
+    F_key,
+    SigF_key,
+    output_columns=None,
+    dropna=True,
+    inplace=False,
+):
+    """
+    Back-calculate intensities and approximate intensity error estimates from
+    structure factor amplitudes and error estimates
+
+    Intensity computed as I = SigF*SigF + F*F. Intensity error estimate
+    approximated as SigI = abs(2*F*SigF)
+
+    Parameters
+    ----------
+    ds : DataSet
+        Input DataSet containing columns with F_key and SigF_key labels
+    F_key : str
+        Column label for structure factor amplitudes to use in back-calculation
+    SigF_key : str
+        Column label for structure factor error estimates to use in
+        back-calculation
+    output_columns : list or tuple of column names
+        Column labels to be added to ds for recording back-calculated I and
+        SigI, respectively. output_columns must have len=2.
+    dropna : bool
+        Whether to drop reflections with NaNs in F_key or SigF_key
+        columns
+    inplace : bool
+        Whether to modify the DataSet in place or create a copy
+
+    Returns
+    -------
+    DataSet
+        DataSet with 2 additional columns corresponding to back-calculated
+        intensities and intensity error estimates
+
+    """
+
+    if not inplace:
+        ds = ds.copy()
+
+    # Sanitize input or check for invalid reflections
+    if dropna:
+        ds.dropna(subset=[F_key, SigF_key], inplace=True)
+    else:
+        if ds[[F_key, SigF_key]].isna().to_numpy().any():
+            raise ValueError(
+                f"Input {ds.__class__.__name__} contains NaNs "
+                f"in columns '{F_key}' and/or '{SigF_key}'. "
+                f"Please fix these input values, or run with dropna=True"
+            )
+
+    if output_columns is None:
+        output_columns = ["I_back", "SigI_back"]
```

Comment:
The `back` suffix is a bit confusing in my mind. Personally, I think it should be `I_calc` and `SigI_calc` instead of `back`. I think `back` implies that they were once intensities to begin with, but that doesn't have to be the case.